src/toil/test/src/jobServiceTest.py

```diff
@@ -109,7 +109,7 @@ def runToil(self, rootJob, retryCount=1, badWorker=0.5, badWorkedFailInterval=0.
         options.logLevel = "INFO"
         
         options.retryCount = retryCount
-        #options.clean = "never"
+        options.clean = "never"
```

Comment:
Ah it is a. My code did not handle 'onsuccess' case correctly. Jobs that raised exceptions were being cleaned up inappropriately.